### PR TITLE
feature/be store provisions integration

### PR DESCRIPTION
**Pull request by UnSk1LLe**
# Description

integration of store provisions with store-synchronization and inventory recalculation system

## Issue ticket link and Sprint

[Link to Asana](https://app.asana.com/1/1206511942979507/project/1209763485270386/task/1209876249859031?focus=true)

PI 2, Sprint 2

## Type of change

Please delete options that are not relevant.

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [X]  New feature (non-breaking change which adds functionality)
- [X]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X]  This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ]  Unit Testing
- [ ]  Integration Testing
- [ ]  E2E Testing
- [X] Manual Testing
# Checklist:

- [X]  My code follows the style guidelines of this project
- [ ]  I have made corresponding changes to the documentation
- [ ]  My changes generate no new warnings
- [ ]  I have added tests that prove my fix is effective or that my feature works
- [ ]  New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by DiarCode** [comment]
1. Fix the merge conflict
2. Test entire workflow with edge cases

**Comment by DiarCode** [comment]
Fix the CI on backend

**Review by DiarCode — CHANGES_REQUESTED** [review]

File: backend/internal/data/store.data.go [hunk 1]
```diff
@@ -107,7 +108,8 @@ type StoreProvision struct {
 	BaseEntity
 	ProvisionID               uint                       `gorm:"not null;index"`
 	Provision                 Provision                  `gorm:"foreignKey:ProvisionID;constraint:OnDelete:CASCADE"`
-	Volume                    float64                    `gorm:"type:decimal(10,2);not null;check:volume > 0" sort:"volume"`
+	Volume                    float64                    `gorm:"type:decimal(10,2);not null" sort:"volume"`
+	InitialVolume             float64                    `gorm:"type:decimal(10,2);not null;check:initial_volume > 0" sort:"initialVolume"`
 	Status                    StoreProvisionStatus       `gorm:"not null" sort:"status"`
```
[the remark this answers is not in the document]
Comment (UnSk1LLe) [reply backend/internal/data/store.data.go:113]: all statuses in are used in DB. But we have additional filtration by expiration date to show the actual state of the StoreProvision, because the only way(for now) for StoreProvision to get expired, is to be caught by notification cronJob that runs every 30mins and notifies about expired StoreProvisions and updates their statuses to "EXPIRED". This doensnt conflict with recalculation logic, since it considers only "COMPLETED" StoreProvisions that are not yet expired

File: backend/internal/modules/provisions/storeProvisions/storeProvisions.repository.go [hunk 2]
```diff
@@ -221,3 +276,25 @@ func (r *storeProvisionRepository) CountStoreProvisionsToday(storeID, provisionI
...
+	}
+
+	var storeProvisionList []data.StoreProvision
+
+	query := r.db.Model(&data.StoreProvision{}).
+		Preload("Provision.Unit").
+		Preload("Store").
+		Where("store_id = ?", storeID).
+		Where("status = ?", data.STORE_PROVISION_STATUS_COMPLETED).
+		Where("expires_at <= ?", time.Now())
```
Comment (DiarCode) [inline backend/internal/modules/provisions/storeProvisions/storeProvisions.repository.go:292]: Here you are not using time.Now().UTC() that will cause improper filtering. But inside your cron task you use  time.Now().UTC(), this will may lead to inconsistency.

Also check other queries with time.Now() or other datetime to check if you use UTC timezone

File: backend/internal/scheduler/storeProvision.schedules.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+		if err != nil {
+			tasks.logger.Errorf("Failed to fetch stock list for store %d: %v", store.ID, err)
+			continue
+		}
+
+		var expiredStoreProvisionIDs []uint
+
+		provisionIDsToRecalculate := make(map[uint]struct{})
+		for _, storeProvision := range storeProvisionList {
+			if storeProvision.Status == data.STORE_PROVISION_STATUS_COMPLETED &&
```
Comment (DiarCode) [inline backend/internal/scheduler/storeProvision.schedules.go:67]: You dont need this check since you are fetching the data with same WHERE CLAUSE. You either how to properly name the function that fits into condition like GETCompletedAndNotExpiredStoreProvisions() to make it clear and don't make redundant checks

File: backend/internal/scheduler/storeProvision.schedules.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	tasks.logger.Info("Running CheckStoreProvisionNotifications...")
+
+	storesList, err := tasks.storeService.GetAllStoresForNotifications()
+	if err != nil {
+		tasks.logger.Errorf("Failed to fetch stores: %v", err)
+		return
+	}
+
+	for _, store := range storesList {
+		processedProvisions := make(map[uint]bool)
```
Comment (DiarCode) [inline backend/internal/scheduler/storeProvision.schedules.go:55]: This map is not used anywhere, so what is purpose of it

File: backend/internal/scheduler/storeProvision.schedules.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+		storeProvisionService:     storeProvisionService,
+		storeProvisionRepo:        storeProvisionRepo,
+		storeInventoryManagerRepo: storeInventoryManagerRepo,
+		storeService:              storeService,
+		notificationService:       notificationService,
+		logger:                    logger,
+	}
+}
+
+func (tasks *StoreProvisionCronTasks) CheckStoreProvisionNotifications() {
```
Comment (DiarCode) [inline backend/internal/scheduler/storeProvision.schedules.go:45]: Overall this function looks ambigious and needs to be divided into helper functions

File: backend/migrations/000009_provision_improvements.up.sql [hunk 6]
```diff
@@ -0,0 +1,7 @@
+ALTER TABLE provisions
```
Comment (DiarCode) [inline backend/migrations/000009_provision_improvements.up.sql:1]: As our data grows, adding indexes on our most-used fields can help keep search times fast. However, let's be cautious not to overdo it—too many indexes can hurt write performance. For a good balance, check out this Cybertec article on PostgreSQL indexing best practices. [ARTICLE](https://www.cybertec-postgresql.com/en/speeding-up-index-creation-in-postgresql/)

**Review by DiarCode — CHANGES_REQUESTED** [review]
